### PR TITLE
feat: upgrade all dependencies and add new consistent react import rule

### PR DESCRIPTION
**Pull request by SimeonC**
<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @tablecheck/babel-preset@1.1.0-canary.47.2032631171.0
  npm install @tablecheck/codemods@1.1.0-canary.47.2032631171.0
  npm install @tablecheck/commitlint-config@1.1.0-canary.47.2032631171.0
  npm install @tablecheck/eslint-config@1.7.0-canary.47.2032631171.0
  npm install @tablecheck/eslint-plugin@1.2.0-canary.47.2032631171.0
  npm install @tablecheck/scripts@1.11.0-canary.47.2032631171.0
  npm install @tablecheck/semantic-release-config@2.5.0-canary.47.2032631171.0
  npm install @tablecheck/scripts-utils@1.1.0-canary.47.2032631171.0
  # or 
  yarn add @tablecheck/babel-preset@1.1.0-canary.47.2032631171.0
  yarn add @tablecheck/codemods@1.1.0-canary.47.2032631171.0
  yarn add @tablecheck/commitlint-config@1.1.0-canary.47.2032631171.0
  yarn add @tablecheck/eslint-config@1.7.0-canary.47.2032631171.0
  yarn add @tablecheck/eslint-plugin@1.2.0-canary.47.2032631171.0
  yarn add @tablecheck/scripts@1.11.0-canary.47.2032631171.0
  yarn add @tablecheck/semantic-release-config@2.5.0-canary.47.2032631171.0
  yarn add @tablecheck/scripts-utils@1.1.0-canary.47.2032631171.0
  ```
</details>

### CODE REVIEW

File: packages/scripts/config/eslintStylishFormatter.js [hunk 1]
```diff
@@ -1,10 +1,120 @@
...
+ */
+function pluralize(word, count) {
+  return count === 1 ? word : `${word}s`;
+}
+
+//------------------------------------------------------------------------------
+// Public Interface
+//------------------------------------------------------------------------------
+
+function stylishFormatter(results) {
```
Comment (SimeonC) [inline packages/scripts/config/eslintStylishFormatter.js:27]: This whole section used to be imported from eslint, but they changed their build process and we can't import it directly anymore so it's copied here.

**Review by daniellizik — APPROVED** [review]

File: packages/eslint-config/rules/react.js [hunk 2]
```diff
@@ -40,5 +40,7 @@ module.exports = {
     'always',
     { ignoreClassFields: true }
   ],
-  'react/no-find-dom-node': 'error'
+  'react/no-find-dom-node': 'error',
+  'react/jsx-no-useless-fragment': ['error', { allowExpressions: true }],
+  '@tablecheck/consistant-react-import': 'error'
```
Comment (gazpachu) [inline packages/eslint-config/rules/react.js:45]: Typo: `consistant`

File: packages/eslint-plugin/index.js [hunk 3]
```diff
@@ -1,6 +1,7 @@
 module.exports = {
   rules: {
     'forbidden-imports': require('./src/forbiddenImports'),
-    'max-mixed-css': require('./src/maxMixedCss')
+    'max-mixed-css': require('./src/maxMixedCss'),
+    'consistant-react-import': require('./src/consistentReactImport')
```
Comment (gazpachu) [inline packages/eslint-plugin/index.js:5]: Here as well

File: packages/scripts/config/eslintStylishFormatter.js (hunk 1 above)
Comment (wahidfarid) [reply packages/scripts/config/eslintStylishFormatter.js:27]: i suppose copying that comment into the code would make it clearer for anyone changing this in the future

File: packages/eslint-config/rules/react.js (hunk 2 above)
Comment (SimeonC) [reply packages/eslint-config/rules/react.js]: Thanks, fixed.

File: packages/eslint-plugin/index.js (hunk 3 above)
Comment (SimeonC) [reply packages/eslint-plugin/index.js]: Thanks, fixed.

**Review by gazpachu — APPROVED** [review]

**Review by wahidfarid — COMMENTED** [review]
Out of scope of this PR, but trying to understand something :bow: 

judging by the amount of packages that use `^` it's an intentional decision. If we are always using explicit version numbers in our apps to avoid breakage, shouldn't we do the same in underlying packages like the tc react system? :thinking: otherwise possible breaking changes can propagate from here into projects that use it

**Review by daniellizik — APPROVED** [review]

**Comment by SimeonC** [comment]
> Out of scope of this PR, but trying to understand something 🙇
> 
> judging by the amount of packages that use `^` it's an intentional decision. If we are always using explicit version numbers in our apps to avoid breakage, shouldn't we do the same in underlying packages like the tc react system? 🤔 otherwise possible breaking changes can propagate from here into projects that use it

We used to use explicit version numbers here too for that reason, but normally we found that that approach made it much harder to get small updates out for security or similar as we'd have to jump through the extra hoops. We decided (I decided?) that the security level stuff is handled by auditjs nicely. As for breaking changes - the versions get locked in package-lock.json files of the projects and with newer versions of npm don't update sub-deps unless you forcibly update them.

TLDR; the risk was determined to be smaller than the effort needed to constantly update them if needed.
